src/Dialogs/SyncWarningDialog.vala

```diff
@@ -26,44 +26,54 @@
  * Authored by: Scott Ringwelski <sgringwe@mtu.edu>
  */
 
-public class Noise.SyncWarningDialog : Gtk.Window {
+public class Noise.SyncWarningDialog : Gtk.Dialog {
+    private enum ResponseId {
+        IMPORT_MEDIA = 1,
+        CONTINUE,
+        STOP
+    }
+
     Device d;
```

Comment:
Make this a construction property and rename appropriately.  e.g. public Device device {get; construct;}